Civi/API/Api4SelectQuery.php

```diff
@@ -147,4 +155,92 @@ protected function getField($fieldName) {
     return NULL;
   }
 
+  protected function buildSelectFields() {
+    parent::buildSelectFields();
+    foreach ($this->select as $selectAlias) {
+      $alreadyAdded = in_array($selectAlias, $this->selectFields);
+      $containsDot = strpos($selectAlias, '.') !== FALSE;
+      if ($alreadyAdded || !$containsDot) {
+        continue;
+      }
+
+      $customFieldData = $this->addDotNotationCustomField($selectAlias);
+
+      if (!$customFieldData) {
+        continue;
+      }
+
+      $tableAlias = $customFieldData[0];
+      $columnName = $customFieldData[1];
+      $field = sprintf('%s.%s', $tableAlias, $columnName);
+
+      $this->selectFields[$field] = $selectAlias;
+    }
+  }
+
+  /**
+   * @param $customField
+   *
+   * @return array|null
+   */
+  protected function addDotNotationCustomField($customField) {
+    list($groupName, $fieldName) = explode('.', $customField);
+
+    $customGroup = $this->getCustomGroup($this->entity, $groupName);
```

Comment:
Careful here. The custom group `extends` property is a little odd - it can be, for example "Contact" "Individual" "Organization" or "Household". Those are all the same entity.